modules/simulation/src/main/resources/schemas/neurosciencegraph/simulation/bluepyoptrun/v0.1.0.json

```diff
@@ -0,0 +1,68 @@
+{
+    "@context": [
+        "{{base}}/contexts/neurosciencegraph/core/schema/v0.1.0",
+        {
+            "this": "{{base}}/schemas/thalamusproject/simulation/bluepyoptrun/v0.1.0/shapes/"
+        }
+    ],
+    "@type": "nxv:Schema",
+    "imports": [
+        "{{base}}/schemas/neurosciencegraph/commons/entity/v1.0.0"
+    ],
+    "shapes": [
+        {
+            "@id": "this:BluePyOptRunShape",
+            "@type": "sh:NodeShape",
+            "label": "BluePyOpt run shape",
+            "comment": "This entity tracks a run of the BluePyOpt software. It registers can be used to register all input and outputs of the run.",
+             "targetClass": "nsg:BluePyOptRun",
+             "nodeKind": "sh:BlankNodeOrIRI",
+            "and": [
+                {
+                    "node": "{{base}}/schemas/neurosciencegraph/commons/entity/v1.0.0/shapes/EntityShape"
+                },
+                {
+                    "path": "nsg:gitHash",
+                    "description": "The git hash of the thalamus repository in which BluePyOpt is being run",
```

Comment:
What is the relevance of thalamus here?

Reply:
Good question. The thing is that this schema is very coupled to the workflow of BBP's Thalamus team. This gitHash simply stores the revision of their workflow.
This makes me wonder:
1) Does this schema have its place in Neuroshape ?
or
2) Should I simply remove the mention to "thalamus" in the description ?

Reply:
I think this would need to be generalized to be usable for all BluePyOpt users, not just one specific team.

Reply:
This has been discussed with @wvangeit (BlueByOpt author) when I started working on this task.

Being an library and not a script that is called with specific arguments, BlueByOpt input can only be captured by capturing the whole content of the code calling it. It will not always be possible to dissociate the input data from the logic by which they are collected. So I don't think generalizing the schema to all BluePyOpt users is doable.
That being said, if you would tell me that this schema is not generic enough to be part of NeuroShape I would totally understand.

Reply:
In the context of the Thalamus workflow the git repository is not a changing value. There is no point in storing it in each node.

Reply:
Taking a long-term perspective, the Git repository might change even for the Thalamus workflow. Adding this to the schema would certainly be a good step in generalizing this, at the cost of adding a bit of redundancy.

Reply:
@wizmer at least for now I guess we could say that these constraints are more less shared among the single cell optimizations.

One more thing though. I would say that the 'BluePyOptRun' name is way to general. BluePyOpt can and is used for several use cases, single cell optimization being one of them. I guess finding a specific enough name will be a general problem with all these entities. Maybe something like SingleCellBluePyOptRun ? Of course, as already mentioned, this specific way of configuring bluepyopt is also specific to e.g. thalamus/sscx atm, not sure if we should also include that in the name in some way.

Reply:
Thanks for the clarification @wvangeit.

Concerning the removal of the gitHash attribute, I have created another PR #202 that is a schema draft for storing information about a software revision (git url, commit, version). It could be useful for the present schema so I think we should start by discussing it.